chain/chain/src/chain.rs

```diff
@@ -4471,8 +4471,9 @@ impl Chain {
         epoch_sync_info: &EpochSyncInfo,
     ) -> Result<(), EpochSyncInfoError> {
         let store = self.store().store().clone();
-        let mut chain_update = self.chain_update();
-        let mut store_update = chain_update.chain_store_update.store().store_update();
+        let epoch_manager = self.epoch_manager.clone();
+        let mut chain_store_update = self.store.store_update();
+        let mut store_update = store.store_update();
```

Comment:
Aah this is super confusing. `store` can be ChainStore or Store depending on whether it comes from self.store or not.

Reply:
In the future we might want to rename all references to `store` of type ChainStore to `chain_store` maybe?

Reply:
yeah, I agree, `chain_store` would be better

Reply:
https://github.com/near/nearcore/pull/10392